src/main/java/com/shoppinglist/shoppinglist/ShoppingListController.java

```diff
@@ -74,9 +83,10 @@ public ListItem fetchListItem(@PathVariable int id) {
     @PostMapping(path = "/post", consumes = "application/json", produces = "application/json")
     public ListItem createListItem(@RequestBody ListItem item) {
         try {
-            return shoppingListService.save(item);
+            return shoppingListService.saveItem(item);
         } catch (IllegalArgumentException e) {
             e.fillInStackTrace();
+            logger.error("Error while creating item: {}", e.getMessage()); // Log the error

```

Comment:
I added logging when exceptions are thrown to track the problem in the application, this ensures that error information is retained for later review.
https://www.baeldung.com/java-clean-code